packages/node-utils/src/http.ts

```diff
@@ -306,6 +390,61 @@ export const allowOrigins =
                 logger,
             })
         ) {
-            next();
+            next(request, _response);
         }
     };
+
+export const parseBodyTextHelper = (request: Request) =>
+    new Promise<string>(resolve => {
+        const tmp: any[] = [];
+        request
+            .on('data', chunk => {
+                tmp.push(chunk);
+            })
+            .on('end', () => {
+                const body = Buffer.concat(tmp).toString();
+                // at this point, `body` has the entire request body stored in it as a string
+                resolve(body);
+            });
+    });
+
+/**
+ * set request.body as parsed JSON
+ */
+export const parseBodyJSON = <R extends RequestWithParams>(
+    request: R,
+    response: Response,
+    next: Next<R & { body: Record<string, any> }>,
+) => {
+    type RequestWithBodyJSON = R & { body: Record<string, any> };
+
+    parseBodyTextHelper(request).then(body => {
+        try {
+            const requestWithBody: RequestWithBodyJSON = {
+                ...request,
+                body: JSON.parse(body),
+            };
+            next(requestWithBody, response);
+        } catch (error) {
+            const requestWithBody: RequestWithBodyJSON = { ...request, body: {} };
+            next(requestWithBody, response);
+        }
+    });
+};
+
+/**
+ * set request.body as string
+ */
+export const parseBodyText = <R extends RequestWithParams>(
+    request: R,
+    response: Response,
+    next: Next<R & { body: string }>,
+) => {
+    parseBodyTextHelper(request).then(body => {
+        const requestWithBody: R & { body: string } = {
+            ...request,
+            body,
+        };
+        next(requestWithBody, response);
+    });
```

Comment:
I'm sorry, I just like short code. Take it or not.
```suggestion
    parseBodyTextHelper(request).then(body => {
        next({ ...request, body }, response);
    });
```

Reply:
[9e5ad43](https://github.com/trezor/trezor-suite/pull/9985/commits/9e5ad43d856d6d2dbed31bb60e1177bb7210f7ac)